### DISCUSSION TITLE
Help! Saving as PSD leaves layer opacity of 0%???

### DISCUSSION THREAD

**Discussion by bioworldmerch** [ImageMagick/ImageMagick#5701] [Help] [answered]
Not sure why Imagemagick is altering the Layer Opacity when running the following command to eliminate transparency around edges. When it is completed, I end up with a one layer file with 0% Opacity. (100% Transparent) when opened in Photoshop.

Any ideas??

`magick infile.psd -units pixelsperinch -density 300 -channel A -threshold 33% outfile.psd`

**Comment by fmw42** [comment]
Can you post your image?  Have you tried turning on all the channels before writing the output.

`magick infile.psd -units pixelsperinch -density 300 -channel A -threshold 33% -channel RGBA outfile.psd
`

**Reply by bioworldmerch** [reply]
they are all on and layers are all 100% opaque. I think it's either a missing portion of command or a bug in IM dealing with PSD. Here's the image...

Thanks again for your response!
[testimage.zip](https://github.com/ImageMagick/ImageMagick/files/9883029/testimage.zip)

**Reply by fmw42** [reply]
I do not understand. When you threshold the opacity/alpha channel, you make it binary. So that removes your anti-aliasing (smooth edges) and causes stair-casing. The command works fine for me doing just that.  Perhaps you want some other behavior and will need a different command for that.  Please explain in more detail what result your desire.

`magick testimage.psd -units pixelsperinch -density 300 -channel A -threshold 33% -channel RGBA testimage_out.psd`

[testimage_out.psd.zip](https://github.com/ImageMagick/ImageMagick/files/9883092/testimage_out.psd.zip)

**Reply by bioworldmerch** [reply]
I want to binary the transparency and save the file out as psd. currently if I run my command or yours the alpha is correct BUT the PSD Layer becomes Opacity of 0% see the attached image opened in Photoshop.

<img width="1474" alt="Screen Shot 2022-10-27 at 5 21 50 PM" src="https://user-images.githubusercontent.com/84971106/198401436-2be22c6b-f0bf-48b6-96b5-8f03d342f665.png">

**Reply by fmw42** [reply]
I do not understand what you are showing me. Please clarify. Is this the image or just the alpha channel. Where is the rest of the image?

This was my output as shown in Photoshop.

![Screen Shot 2022-10-27 at 2 35 23 PM](https://user-images.githubusercontent.com/13458218/198402935-56b6d928-5397-41c8-9263-774d48ba9068.png)

Looks fine to me.  Is that not what you get?

**Reply by bioworldmerch** [reply]
No when I run it the Opacity of Layer 1 ends up 0%. When I open the file I sent, it looks similar to your image (normal less the thresholded transparency) then after running command it opens like mine with your command or mine.

**Reply by fmw42** [reply]
Sorry, I do not understand.  What do you mean by “run the opacity”? How are you measuring it?  Is your resulting image totally transparent so that you do not see your figure?

**Reply by fmw42** [reply]
What is your IM version and platform?  Are you sure it is installed correctly?  What do you get from 

`magick -version`

**Reply by bioworldmerch** [reply]
Odd it also comes in with the layer locked like yours but not with 100% opacity. Im running Photoshop 2023 as well and Imagemagick ImageMagick 7.1.0-45. You don't have a .PSD Library for Imagemagick right? Mine is Q32.

**Reply by bioworldmerch** [reply]
In photoshop you can change % of opacity of layer similar to effects like multiply etc. Mine ends up 0% when opening the outfile from Imagemagick. I did recently have my Mac laptop screen go out so I switched to M1 laptop and migrated. Was surprised all ran fine. I don't think it is related though...unless you do. Here's -version:

Version: ImageMagick 7.1.0-45 Q32-HDRI x86_64 20319 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1) 
Delegates (built-in): bzlib freetype heic jng jp2 jpeg lcms lqr ltdl lzma openexr png tiff webp xml zlib
Compiler: gcc (4.2)

**Reply by fmw42** [reply]
Where are you controlling or viewing the opacity value.  The value shown in the window in  my screen grab is the opacity for the blend.  I have not idea why you are getting 0% and I see 100%.  This would seem to be an issue with Photoshop or some setting you have.  What version of PS are you using?  How does it display in Imagemagick?

**Reply by fmw42** [reply]
Perhaps install the latest Imagemagick and see if that fixes it.

**Reply by bioworldmerch** [reply]
I reinstalled IM and it seems to have resolved the issue. Thank you for your patience!

**Comment by bioworldmerch** [comment, marked as answer]
As @fmw42 suggested install/reinstall latest resolved issues. Thanks again!
